src/synapse-plugins/converter-plugin/converter-data.vala

```diff
@@ -0,0 +1,203 @@
+/*
+* Copyright (c) 2022 elementary LLC.
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public
+* License as published by the Free Software Foundation; either
+* version 2 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* General Public License for more details.
+*
+* You should have received a copy of the GNU General Public
+* License along with this program; if not, write to the
+* Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+* Boston, MA 02110-1301 USA
+*
+* Authored by: Jeremy Wootten <jeremywootten@gmail.com>
+*/
+
+namespace Synapse {
+    const string MASS = "unit of mass";
+    const string LENGTH = "unit of length";
+    const string VOLUME = "unit of volume";
+    const string TIME = "unit of time";
+
+    const SIPrefix [] PREFIXES = {
+        {"yotta", "Y", 1E24},
+        {"zetta", "Z", 1E21},
+        {"exa", "E", 1E18},
+        {"peta", "P", 1E15},
+        {"tera", "T", 1E12},
+        {"giga", "G", 1E9},
+        {"mega", "M", 1E6},
+        {"kilo", "k", 1000},
+        {"hecto", "h", 100},
+        {"deca", "da", 10},
+        {"deci", "d", 0.1},
+        {"centi", "c", 0.01},
+        {"milli", "m", 0.001},
+        {"micro", "u", 1E-6},
+        {"nano", "n", 1E-9},
+        {"pico", "p", 1E-12},
+        {"femto", "f", 1E-15},
+        {"atto", "a", 1E-18},
+        {"zepto", "z", 1E-21},
+        {"yocto", "y", 1E-24}
+    };
+
+    enum UnitSystem {
+        SI,
+        IMPERIAL, // Where definition same in US and UK or there is no equivalent in other country and defined in terms of an IMPERIAL measure
+        IMPERIAL_UK, // Where definition differs from US or not defined in IMPERIAL measure
+        IMPERIAL_US, // Where definition differs from UK or not defined in IMPERIAL measure
+    }
+
+    enum UnitType { // UnitTypes are not interconvertible with dimensions
+        MASS,
+        DIMENSION, // Length, Area, Volume are interconvertible with dimensions (L^1, L^2, L^3)
+        TIME,
+        VELOCITY
+    }
+
+    struct Unit {
+        public UnitType type; // Units of different types cannot be interconverted
+        public UnitSystem system; // e.g. SI, Imperial, Chinese
+        public string uid; // Unique identifier for the purposes of the converter - not official
```

Comment:
Just that it does not have to correspond to anything used in the real world (although it will be recognized if entered).  The important thing is that it is unique.